lib/storage/bucket.js

```diff
@@ -198,6 +198,10 @@ Bucket.prototype.delete = function(callback) {
  * var file = bucket.file('my-existing-file.png');
  */
 Bucket.prototype.file = function(name) {
+  if (!name) {
```

Comment:
Checking for the presence of a Bucket is out since my first reply in the issue. It's not needed, since the only code that would be caught by that error is our own code (not a user's), and we can better prevent that error from happening by our test suite, rather than a user seeing an error that makes no sense to them.

I said above it makes sense to have the error handling done in Bucket#file since, if we're being modular, in order for that method to successfully execute, it needs a name, because we, as the authors of this library, know that our File object needs a name. We can throw in either place, it really doesn't matter, and since we each thought differently, I thought @coderaiser could save us the trouble and decide for us. If you feel more strongly that everything is fine the way it is now, feel free to close this out.

(Either way, thanks @coderaiser for helping us out and your patience!)